pgamit/pyETM.py

```diff
@@ -2770,14 +2862,15 @@ def nearestPD(self, A):
             return A3
 
         spacing = np.spacing(np.linalg.norm(A))
-        # The above is different from [1]. It appears that MATLAB's `chol` Cholesky
-        # decomposition will accept matrixes with exactly 0-eigenvalue, whereas
-        # Numpy's will not. So where [1] uses `eps(mineig)` (where `eps` is Matlab
-        # for `np.spacing`), we use the above definition. CAVEAT: our `spacing`
-        # will be much larger than [1]'s `eps(mineig)`, since `mineig` is usually on
-        # the order of 1e-16, and `eps(1e-16)` is on the order of 1e-34, whereas
-        # `spacing` will, for Gaussian random matrixes of small dimension, be on
-        # othe order of 1e-16. In practice, both ways converge, as the unit test
```

Comment:
@demiangomez not a comment on @pdsmith90 pep8 fixes, but a comment on the numerics-- numpy [cholesky](https://numpy.org/doc/stable/reference/generated/numpy.linalg.cholesky.html) decomposition will accept any valid positive semi-definite input, and is much faster than the `svd` that is being used here. If you want / need to still output decomposition values for matrices which aren't positive semi-definite, I believe that you can use `multivariate_normal`, which will default to `cholesky` if possible (for speed), but then fall back to `svd` if `cholesky` raises a `RuntimeWarning: covariance is not positive-semidefinite.` warning during runtime.

Might speed up your runtime, but also not sure how much this code is dominating runtime for you at present.